modules/json/format-json.c

```diff
@@ -234,6 +234,24 @@ tf_json_append_list(const gchar *name, const gchar *value, gsize value_len,
   g_string_append_c(state->buffer, ']');
 }
 
+/* RFC8259 numbers: '+' sign, leading zeros are not allowed. */
+static inline void
+tf_json_append_double(const gchar *name, gdouble d, json_state_t *state)
+{
+  gchar double_buf[G_ASCII_DTOSTR_BUF_SIZE];
+  g_ascii_dtostr(double_buf, G_N_ELEMENTS(double_buf), d);
+  tf_json_append_value(name, double_buf, -1, state, FALSE);
+}
+
+static inline void
+tf_json_append_int(const gchar *name, gint64 i, json_state_t *state)
+{
+  gchar int_buf[32];
+  g_snprintf(int_buf, G_N_ELEMENTS(int_buf), "%"G_GINT64_FORMAT, i);
```

Comment:
could we use something other than sprintf here? I am not sure how much integer fields we can expect, but sprintf is really slow.

Reply:
This is an interesting post about itoa() implementations:
https://jk-jeon.github.io/posts/2022/02/jeaiii-algorithm/

Since itoa is not portable, we could use our own, which does not have to be that complex.
The naive or the 2-digit-per-div method would be probably enough for us.